QMChatService/QMChatService/QMChatService.m

```diff
@@ -1173,65 +1173,88 @@ - (void)fetchDialogsUpdatedFromDate:(NSDate *)date andPageLimit:(NSUInteger)limi
 #pragma mark - Send messages
 
 - (void)sendMessage:(QBChatMessage *)message
+			   type:(QMMessageType)type
+		   toDialog:(QBChatDialog *)dialog
+		withoutJoin:(BOOL)withoutJoin
+	  saveToHistory:(BOOL)saveToHistory
+	  saveToStorage:(BOOL)saveToStorage
+		 completion:(QBChatCompletionBlock)completion
+{
+	message.dateSent = [NSDate date];
+	
+	//Save to history
+	if (saveToHistory) {
+		message.saveToHistory = kChatServiceSaveToHistoryTrue;
+	}
+	//Set message type
+	if (type != QMMessageTypeText) {
+		message.messageType = type;
+	}
+	
+	QBUUser *currentUser = self.serviceManager.currentUser;
+	
+	if (dialog.type == QBChatDialogTypePrivate) {
+		message.recipientID = dialog.recipientID;
+		message.markable = YES;
+	}
+	
+	message.senderID = currentUser.ID;
+	message.dialogID = dialog.ID;
+	
+	__weak __typeof(self)weakSelf = self;
+	void(^sendMessageCompletionBlock)(NSError *) = ^(NSError *error) {
+		__typeof(weakSelf)strongSelf = weakSelf;
+		
+		if (error == nil && saveToStorage) {
+			
+			// there is a case when message that was returned from server (Group dialogs)
+			// will be handled faster then this completion block been fired
+			// therefore there is no need to add local message to memory storage, while server
+			// up-to-date one is already there
+			BOOL messageExists = [strongSelf.messagesMemoryStorage isMessageExistent:message forDialogID:message.dialogID];
```

Comment:
@hausegger: [See this comment](https://github.com/mysugr/quickblox-services-ios/pull/1#discussion_r78276835)